contracts/modules/cover/Cover.sol

```diff
@@ -0,0 +1,773 @@
+
+    uint allocationCount = allocations.length;
+    for (uint i = 0; i < allocationCount; i++) {
+
+      PoolAllocation storage allocation = allocations[i];
+      IStakingPool _stakingPool = stakingPool(allocation.poolId);
+
+      uint nxmBurned = allocation.coverAmountInNXM
+        * burnAmount / segment.amount
+        * GLOBAL_CAPACITY_DENOMINATOR / globalCapacityRatio;
+
+      _stakingPool.burnStake(cover.productId, segment.start, segment.period, nxmBurned);
+
+      uint payoutAmountInNXM = allocation.coverAmountInNXM * burnAmount / segment.amount;
+      allocation.coverAmountInNXM -= SafeUintCast.toUint96(payoutAmountInNXM);
+    }
+
+    return owner;
+  }
+
+  function transferCovers(address from, address to, uint256[] calldata coverIds) external override {
+    require(
+      msg.sender == internalContracts[uint(ID.MR)],
+      "Cover: Only MemberRoles is permitted to use operator transfer"
+    );
+
+    ICoverNFT coverNFTContract = ICoverNFT(coverNFT);
+    for (uint256 i = 0; i < coverIds.length; i++) {
+      coverNFTContract.operatorTransferFrom(from, to, coverIds[i]);
+    }
+  }
+
+
+  function retrievePayment(
+    uint premium,
+    uint8 paymentAsset,
+    uint16 commissionRatio,
+    address commissionDestination
+  ) internal {
+
+    // add commission
+    uint commission = premium * commissionRatio / COMMISSION_DENOMINATOR;
+
+    if (paymentAsset == 0) {
+
+      uint premiumWithCommission = premium + commission;
+      require(msg.value >= premiumWithCommission, "Cover: Insufficient ETH sent");
+
+      uint remainder = msg.value - premiumWithCommission;
+
+      if (remainder > 0) {
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool ok, /* data */) = address(msg.sender).call{value: remainder}("");
+        require(ok, "Cover: Returning ETH remainder to sender failed.");
+      }
+
+      // send commission
+      if (commission > 0) {
+        (bool ok, /* data */) = address(commissionDestination).call{value: commission}("");
+        require(ok, "Cover: Sending ETH to commission destination failed.");
+      }
+
+      return;
+    }
+
+    IPool _pool = pool();
+
+    (
+    address payoutAsset,
+    /*uint8 decimals*/
+    ) = _pool.coverAssets(paymentAsset);
+
+    IERC20 token = IERC20(payoutAsset);
+    token.safeTransferFrom(msg.sender, address(_pool), premium);
+
+    if (commission > 0) {
+      token.safeTransferFrom(msg.sender, commissionDestination, commission);
+    }
+  }
+
+  function retrieveNXMPayment(uint price, uint commissionRatio, address commissionDestination) internal {
+
+    ITokenController _tokenController = tokenController();
+
+    if (commissionRatio > 0) {
+      uint commission = price * commissionRatio / COMMISSION_DENOMINATOR;
+      // transfer the commission to the commissionDestination; reverts if commissionDestination is not a member
+      _tokenController.operatorTransfer(msg.sender, commissionDestination, commission);
```

Comment:
Partially bypasses the governance vote lock

Reply:
use 2 hop transfer then?